extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -62,8 +69,41 @@ export class AnalyticsRoot {
     /** @private {?./visibility-manager.VisibilityManager} */
     this.visibilityManager_ = null;
 
+    this.visibilityMangerPromise_ = null;
+
     /** @private {?./scroll-manager.ScrollManager} */
     this.scrollManager_ = null;
+
+    this.usingHostAPIPromise_ = null;
+
+    this.hostVisibilityService_ = null;
+  }
+
+  /**
+   * @return {!Promise<boolean>}
+   */
+  isUsingHostAPI() {
+    if (this.usingHostAPIPromise_) {
+      return this.usingHostAPIPromise_;
+    }
+    if (!HostServices.isAvailable(this.ampdoc)) {
+      this.usingHostAPIPromise_ = Promise.resolve(false);
+    } else {
+      // TODO: Using the visibility service and apply it for all tracking types
+      const promise = HostServices.visibilityForDoc(this.ampdoc);
+      this.usingHostAPIPromise_ = promise.then(visibilityService => {
+        this.hostVisibilityService_ = visibilityService;
+        return true;
+      }).catch(errorCode => {
+        dev().fine(TAG, 'HostServiceError: ' + errorCode);
+        if (errorCode == HostServiceError.MISMATCH) {
+          return false;
+        }
+        // TODO: What to do in this case.
+        throw user().createError('Host API unsupported');
```

Comment:
once switched to the new error handling object, you will read error.fallback to decide if we want to fallback to InOb.

Reply:
Sounds good.

Reply:
Changed to using error.fallback